collated/main/A0093896H.md

```diff
@@ -0,0 +1,2010 @@
+
+
+/**
+ * Change data storage location
+ */
+public class StoreCommand extends Command {
+
+    public static final String COMMAND_WORD = "store";
+
+    public static final String MESSAGE_SUCCESS = "Change storage location !";
+    
+    public static final String MESSAGE_SAVE_ERROR = "Unable to save to new location.";
+    
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Changes the storage location to the location specified by the user. "
+            + "Parameters: NEW_STORAGE_LOCATION\n"
+            + "Example: " + COMMAND_WORD + "data/dodobird.xml";
+
+    private final String location;
+    
+    public StoreCommand(String newLocation) {
+        this.location = newLocation;
+    }
+    
+    @Override
+    public CommandResult execute() {
+        String defaultLocation = config.getToDoListFilePath();
+        String tempLocation = this.location + "/" + config.getToDoListName() + ".xml";
+        
+        if (FileUtil.isFilenameValid(tempLocation)) {
+            try {
+                config.setToDoListFilePath(tempLocation);
+                ConfigUtil.saveConfig(config, Config.USER_CONFIG_FILE);
+                storage.setToDoListFilePath(tempLocation);
+                storage.saveToDoList(model.getToDoList());
+                return new CommandResult(MESSAGE_SUCCESS);
+            } catch (IOException e){
+                config.setToDoListFilePath(defaultLocation);
+                storage.setToDoListFilePath(defaultLocation);
+                return new CommandResult(MESSAGE_SAVE_ERROR);
+            }
+        } else {
+            return new CommandResult(MESSAGE_SAVE_ERROR);
+        }
+          
+    }
+        
+   
+}
+```
+###### \java\seedu\todo\logic\commands\TagCommand.java
+``` java
+package seedu.todo.logic.commands;
+
+import seedu.todo.commons.core.Messages;
+import seedu.todo.commons.core.UnmodifiableObservableList;
+import seedu.todo.model.tag.Tag;
+import seedu.todo.model.tag.UniqueTagList;
+import seedu.todo.model.tag.UniqueTagList.DuplicateTagException;
+import seedu.todo.model.task.ReadOnlyTask;
+import seedu.todo.model.task.Task;
+import seedu.todo.model.task.UniqueTaskList.TaskNotFoundException;
+
+/**
+ * Tags a task identified using it's last displayed index from the to do list.
+ * with tags
+ */
+public class TagCommand extends Command{
+
+    public static final String COMMAND_WORD = "tag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Tags the task identified by the index number used in the last task listing. "
+            + "Tag name must be unique\n"
+            + "Parameters: INDEX TAGNAME [MORE TAGNAMES]\n"
+            + "Example: " + COMMAND_WORD + " 1 birthday clique";
+
+    public static final String MESSAGE_SUCCESS = "Tagged Task: Name : %1$s";
+
+    public final int targetIndex;
+    public final UniqueTagList tags;
+    
+    public TagCommand(int targetIndex, String tagNames) throws Exception {
```

Comment:
the exception is too generic here; can consider adding something more specific to indicate the kind of exception